react-native/react/menubar/index.js

```diff
@@ -59,13 +59,15 @@ class Menubar extends Component {
 
   checkForFolders () {
     if (this.props.username && this.props.loggedIn && !this.props.loading) {
-      this.setState({loading: true})
-      this.props.favoriteList()
+      setImmediate(() => {
+        this.setState({loading: true})
+        this.props.favoriteList()
+      })
     }
   }
 
   componentWillReceiveProps (nextProps) {
-    if (this.props.folders !== nextProps.folders) {
+    if (this.state.loading && (this.props.folders !== nextProps.folders)) {
```

Comment:
I'm not sure about this logic, it seems :fish: .

Maybe you want `componentDidUpdate` instead?